src/acp/util/setVcAccess.ts

```diff
@@ -63,56 +96,63 @@ export const DEFAULT_VC_MATCHER_NAME = "defaultVcMatcher";
  * @since 1.17.0
  */
 export function setVcAccess(
   resourceWithAcr: WithAccessibleAcr,
-  access: Partial<AccessModes>
+  access: Partial<AccessModes>,
+  options: {
+    inherit: boolean;
+  } = { inherit: false }
 ): WithAccessibleAcr {
   let acr = internal_getAcr(resourceWithAcr);
   const defaultVcPolicyIri = `${getSourceIri(acr)}#${DEFAULT_VC_POLICY_NAME}`;
   const defaultVcMatcherIri = `${getSourceIri(acr)}#${DEFAULT_VC_MATCHER_NAME}`;
 
+  const { policy, matcher } = createVcPolicy(
+    acr,
+    defaultVcPolicyIri,
+    defaultVcMatcherIri,
+    access
+  );
+
   let accessControl = getDefaultAccessControlThing(
     resourceWithAcr,
     "defaultAccessControl"
   );
+  if (!getIriAll(accessControl, ACP.apply).includes(asIri(policy))) {
+    // Case when the ACR Thing existed, but did not include a link to the default Access Control.
+    accessControl = addIri(accessControl, ACP.apply, policy);
+  }
+
+  let memberAccessControl = getDefaultAccessControlThing(
+    resourceWithAcr,
+    "defaultMemberAccessControl"
+  );
 
   let acrThing =
     getAccessControlResourceThing(resourceWithAcr) ??
-    buildThing({ url: getSourceIri(acr) })
-      .addIri(ACP.accessControl, accessControl)
-      .build();
+    createThing({ url: getSourceIri(acr) });
 
   if (!getIriAll(acrThing, ACP.accessControl).includes(asIri(accessControl))) {
     // Case when the ACR Thing existed, but did not include a link to the default Access Control.
     acrThing = addIri(acrThing, ACP.accessControl, accessControl);
   }
 
-  let vcPolicy = getThing(acr, defaultVcPolicyIri);
-  if (vcPolicy === null) {
-    // If the policy does not exist, create it and link the default Access Control to it.
-    vcPolicy = buildThing({ url: defaultVcPolicyIri })
-      .addIri(rdf.type, ACP.Policy)
-      .addIri(ACP.anyOf, defaultVcMatcherIri)
-      .build();
-    accessControl = addIri(accessControl, ACP.apply, vcPolicy);
-  }
-
-  const vcMatcher: Thing =
-    getThing(acr, defaultVcMatcherIri) ??
-    buildThing({ url: defaultVcMatcherIri })
-      .addIri(rdf.type, ACP.Matcher)
-      .addIri(ACP.vc, VC_ACCESS_GRANT)
-      .build();
-
-  const currentModes = getModes(vcPolicy, ACP.allow);
-  // Only change the modes which are set in `access`, and preserve the others.
-  vcPolicy = setModes(vcPolicy, { ...currentModes, ...access }, ACP.allow);
-
   // Write the changed access control, policy and matchers in the ACR
-  acr = [acrThing, accessControl, vcPolicy, vcMatcher].reduce(setThing, acr);
+  acr = [acrThing, accessControl, policy, matcher].reduce(setThing, acr);
+
+  if (options.inherit) {
+    // Add triples to the member access control and link it to the ACR only
+    // if the VC access is recursive.
+    memberAccessControl = addIri(memberAccessControl, ACP.apply, policy);
+    acrThing = addIri(acrThing, ACP.memberAccessControl, memberAccessControl);
+    acr = [acrThing, memberAccessControl].reduce(setThing, acr);
+  }
```

Comment:
The inherit property defaults to true, so maybe you want to mimic that here:
- set allow member all the time if the resource is a container
- do not set allow member if inherit is false

Reply:
The `inherit` property defaults to `true` in Access Grants, so I preferred setting the default behaviour at this layer rather than at the lower ACP layer: https://github.com/inrupt/solid-client-access-grants-js/pull/477/files#diff-ab8c49f7948de39654745d99bd32d00025c021d76c1c9dfdedbbeeda66d48ce1.

Reply:
When looking at the function's signature, I see:

```javascript
  options: {
    inherit: boolean;
  } = { inherit: false }
```

Which means that when this function is called without explicitly saying `inherit: true`, it will default to not setting member access, which is the oposite of what access grant will do. I guess it really depends on what you want and who calls it ^^,

Maybe if we're sure that the caller will always determine whether inheritance must apply or not, the parameter should not be optional?

Reply:
Absolutely: it being recursive by default is something that is a piece of logic specific to access grants. That is why the default behaviour in the Access Grant client is the following: 
```
const updatedResource = acp_ess_2.setVcAccess(resourceInfo, accessMode, {
  inherit: options?.inherit ?? true,
});
```
which results in the VC matcher being applied to the member Access Control. However, my assumption was that we want to keep this default recursive behaviour to the Access Grant layer, and have the lower-level ACP layer to default to a more closed access for VCs unless specified otherwise.

This also keeps the breaking change to the Access Grant layer, where it is aligned with the server behaviour.